File: app/src/main/java/com/kaisa/yql/MainActivity.java
package com.kaisa.yql;

import android.os.Bundle;
import android.util.Log;
import android.view.View;

import com.kaisa.yql.push.TagAliasOperatorHelper;

import java.util.LinkedHashSet;
import java.util.Set;

import androidx.appcompat.app.AppCompatActivity;
import cn.jpush.android.api.JPluginPlatformInterface;


public class MainActivity extends AppCompatActivity implements View.OnClickListener {

    private JPluginPlatformInterface jPluginPlatformInterface;
    private String alias = "yuong1992";

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        jPluginPlatformInterface = new JPluginPlatformInterface(this);
        initView();
    }

    private void initView() {
        findViewById(R.id.btn_register_id).setOnClickListener(this);
        findViewById(R.id.btn_set_alias).setOnClickListener(this);
        findViewById(R.id.btn_delete_alias).setOnClickListener(this);
        findViewById(R.id.btn_set_tag).setOnClickListener(this);
        findViewById(R.id.btn_delete_tag).setOnClickListener(this);
        findViewById(R.id.btn_clear_tag).setOnClickListener(this);
    }

    @Override
    public void onClick(View v) {
        switch (v.getId()) {
            case R.id.btn_register_id:
                String rid = TagAliasOperatorHelper.getInstance().getRegisterId(this);
                Log.e("JPUSH", "rid : " + rid);
                break;
            case R.id.btn_set_alias:
                TagAliasOperatorHelper.getInstance().setAlias(this, alias);
                break;
            case R.id.btn_delete_alias:
                TagAliasOperatorHelper.getInstance().deleteAlias(this);
                break;
            case R.id.btn_set_tag: {
                Set<String> tags = new LinkedHashSet<>();
                tags.add(alias);
                TagAliasOperatorHelper.getInstance().setTag(this, tags);
            }
            break;
            case R.id.btn_delete_tag: {
                Set<String> tags = new LinkedHashSet<>();
                tags.add(alias);
                TagAliasOperatorHelper.getInstance().deleteTag(this, tags);
            }
            break;
            case R.id.btn_clear_tag:
                TagAliasOperatorHelper.getInstance().clearTag(this);
                break;
        }
    }

    @Override
    protected void onStart() {
        super.onStart();
        jPluginPlatformInterface.onStart(this);
    }
}

File: app/src/main/java/com/kaisa/yql/MainActivity.kt
package com.kaisa.yql

import android.os.Bundle
import android.util.Log
import android.view.View
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import cn.jpush.android.api.JPluginPlatformInterface
import com.kaisa.yql.push.TagAliasOperatorHelper
import kotlinx.android.synthetic.main.activity_main.*
import java.util.*

class MainActivity : AppCompatActivity(), View.OnClickListener {
    private var jPluginPlatformInterface: JPluginPlatformInterface? = null
    private val alias = "yuong1992"
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        jPluginPlatformInterface = JPluginPlatformInterface(this)
        initView()
    }

    private fun initView() {
        btn_register_id.setOnClickListener(this)
        btn_set_alias.setOnClickListener(this)
        btn_delete_alias.setOnClickListener(this)
        btn_set_tag.setOnClickListener(this)
        btn_delete_tag.setOnClickListener(this)
        btn_clear_tag.setOnClickListener(this)
    }

    override fun onClick(v: View) {
        when (v.id) {
            R.id.btn_register_id -> {
                val rid = TagAliasOperatorHelper.getInstance().getRegisterId(this)
                Log.e("JPUSH", "rid : $rid")
            }
            R.id.btn_set_alias -> TagAliasOperatorHelper.getInstance().setAlias(this, alias)
            R.id.btn_delete_alias -> TagAliasOperatorHelper.getInstance().deleteAlias(this)
            R.id.btn_set_tag -> {
                val tags: MutableSet<String> = LinkedHashSet()
                tags.add(alias)
                TagAliasOperatorHelper.getInstance().setTag(this, tags)
            }
            R.id.btn_delete_tag -> {
                val tags: MutableSet<String> = LinkedHashSet()
                tags.add(alias)
                TagAliasOperatorHelper.getInstance().deleteTag(this, tags)
            }
            R.id.btn_clear_tag -> TagAliasOperatorHelper.getInstance().clearTag(this)
        }
    }

    override fun onStart() {
        super.onStart()
        jPluginPlatformInterface!!.onStart(this)
    }
}